go-controller/pkg/ovn/controller/admin_network_policy/admin_network_policy.go

```diff
@@ -356,9 +356,13 @@ func (c *Controller) setAdminNetworkPolicy(anp *anpapi.AdminNetworkPolicy) error
 	// We only support priority ranges 0-99 in OVN-K for an ANP even if upstream
 	// supports upto 1000. The 0 (highest) corresponds to 30,000 in OVN world and
 	// 99 (lowest) corresponds to 20,100 in OVN world
+	var errMessageString string
 	if anp.Spec.Priority > OVNKSupportedPriorityUpperBound {
-		klog.Errorf("Error attempting to add ANP %s with priority %d because, "+
+		errMessageString = fmt.Sprintf("Error attempting to add ANP %s with priority %d because, "+
 			"OVNK only supports priority ranges 0-99", anp.Name, anp.Spec.Priority)
+		klog.Errorf(errMessageString)
+		// we can ignore the error if status update doesn't succeed; best effort
+		_ = c.updateANPStatusToNotReady(anp, c.zone, errMessageString)
```

Comment:
hmm I think that would be a bit too high up to put status update specific things... besides `clearAdminNetworkPolicy` doesn't really implement status updates since the ANP object is gone by then and an error during deletion cannot really be reflected in an object that is already gone..

I think this comment is stemming from the duplcation perspective, I will move this inside `syncAdminNetworkPolicy` instead based on the result of `setAdminNetworkPolicy`, hope that's ok?

Reply:
oh wait I remember why I didn't do this.. So its specific to the type of error I am having, example when someone tries to create an ANP with priority 99 -> that is not really an error since its allowed by kapi and is an OVNK specific issue so retrying that error 15 times will not benefit us. So I return nil but convey it to the user through status that the ANP is not ready unless they update it with another lower priority, Same when user creates two ANPs with same priority - those things made me use this logic of having to move it down into `setAdminNetworkPolicy`; open to suggestions...

Reply:
discussed this with Nadia, plan is to define specific error types which we can then filter on the upper level that can help decide if we want to retry or not.

Reply:
this is fixed.